### PR TITLE
Upgrade to dotnet 6

### PR DESCRIPTION
**Pull request by kelseyduffy**

### CODE REVIEW

File: OMFAPI/Program.cs [hunk 1]
```diff
@@ -21,7 +21,8 @@ public static class Program
         // Constant for determining the pause between sending OMF data messages
         private const int SendSleep = 1000;
 
-        private static readonly HttpClient _client = new ();
+        private static readonly List<HttpClient> _clients = new ();
```
Comment (cbardini) [inline OMFAPI/Program.cs:24]: What do you think of just having the client be part of the ```Endpoint``` object? I think it could simply things.

File: OMFAPITests/UnitTest1.cs [hunk 2]
```diff
@@ -101,22 +101,24 @@ private static bool Cleanup()
```
Comment (anton-dahlin) [inline OMFAPITests/UnitTest1.cs:101]: While you're in here, would you mind renaming this file? This isn't actually a unit test, we should either be calling these files Tests or E2ETests imo. I could do this in a separate PR if you want to move on for now.

Comment (kelseyduffy) [reply OMFAPITests/UnitTest1.cs:101]: I agree with this, but I think it's probably better for a separate effort since this is found in almost all of our repos. Most of our files are named UnitTest.cs, but perform E2E tests instead. Doing this separately would make the change more trackable I think.

File: OMFAPI/Program.cs (hunk 1 above)
Comment (kelseyduffy) [reply OMFAPI/Program.cs:24]: That's probably better, but my only reservation is that the test rebuilds the endpoints. So the logic for the cert handler would have to live in both places (not doing that could cause the cert check to fail for the cleanup messages). As it is now, both the test and the main program should be pretty robust in each endpoint knowing its index, and with the List<HttpClient> object staying alive until the whole thing is done, it should safely access the client that it needs each time, implicitly picking up the cert handler callback.

Comment (derekendres) [reply OMFAPI/Program.cs:24]: i like it better on the object too even if that means rebuilding some of the code in the test, because the addition of having to handle the httpclient in the main part of the code feels odd to me.  I would rather have that all handled by the endpoint object.

**Review by anton-dahlin — APPROVED** [review]

File: OMFAPI/Program.cs [hunk 3]
```diff
@@ -61,8 +59,19 @@ public static bool RunMain(bool test = false, Dictionary<string, dynamic> lastSe
                 foreach (Endpoint endpoint in endpoints)
                 {
                     if ((endpoint.VerifySSL is bool boolean) && boolean == false)
+                    {
                         Console.WriteLine("You are not verifying the certificate of the end point.  This is not advised for any system as there are security issues with doing this.");
```
Comment (derekendres) [inline OMFAPI/Program.cs:63]: should we consider making this more clear what endpoint we aren't doing cert checks on

Comment (kelseyduffy) [reply OMFAPI/Program.cs]: agreed, what do you think of the new messaging?

File: OMFAPI/Program.cs [hunk 4]
```diff
@@ -60,14 +60,23 @@ public static bool RunMain(bool test = false, Dictionary<string, dynamic> lastSe
...
 
-                        // Create custom callback to disable certificate checks, create an HttpClient object using it
-                        endpoint.Handler = new ();
-                        endpoint.Handler.ServerCertificateCustomValidationCallback = (sender, certificate, chain, sslPolicyErrors) => true;
-                        endpoint.Client = new (endpoint.Handler);
+                            // Create custom callback to disable certificate checks, create an HttpClient object using it
+                            endpoint.Handler = new ();
+                            endpoint.Handler.ServerCertificateCustomValidationCallback = (sender, certificate, chain, sslPolicyErrors) => true;
+                            endpoint.Client = new (endpoint.Handler);
+                        }
```
Comment (derekendres) [inline OMFAPI/Program.cs:76]: What about EDS?

Comment (kelseyduffy) [reply OMFAPI/Program.cs:76]: I wasn't sure what to do with that one because it's not https. so even if you add a callback that short circuits the verification, it shouldn't even be called in the first place because there's no certificate presented. But I should test it out to verify instead of just assuming. Either way, the property is not in the EDS section of the appsettings placeholder example, so people shouldn't be using it, but since all three share a common endpoint class it is theoretically possible for someone to set it anyway.

File: OMFAPI/Program.cs [hunk 5]
```diff
@@ -61,7 +59,31 @@ public static bool RunMain(bool test = false, Dictionary<string, dynamic> lastSe
                 foreach (Endpoint endpoint in endpoints)
                 {
                     if ((endpoint.VerifySSL is bool boolean) && boolean == false)
-                        Console.WriteLine("You are not verifying the certificate of the end point.  This is not advised for any system as there are security issues with doing this.");
+                    {
+                        if (string.Equals(endpoint.EndpointType, "ADH", StringComparison.OrdinalIgnoreCase))
+                        {
+                            // The certificate check should not fail for AVEVA Data Hub since that certificate is managed by AVEVA. If the certificate verification is failing, please contact Technical Support.
+                            Console.WriteLine($"Certificate verification should not be diabled for AVEVA Data Hub endpoints. The VerifySSL setting will be ignore for namespace {endpoint.NamespaceId}.");
```
Comment (cbardini) [inline OMFAPI/Program.cs:66]: *ignored (this is also in the EDS message)

**Review by cbardini — APPROVED** [review]

File: OMFAPI/Program.cs (hunk 5 above)
Comment (cbardini) [reply OMFAPI/Program.cs:66]: Oops didn't realise the PR was set to automerge.
